rfd/0109-opsgenie-plugin.md

```diff
@@ -0,0 +1,101 @@
+# Opsgenie Integration RFD
+## Required approvers
+
+Engineering:
+Product:
+Security:
+
+## What
+
+This RFD proposes a plugin that allows Teleport to integrate with Opsgenie, allowing access requests to show up as alerts in Opsgenie. This plugin will differ from existing plugins by being part of the Teleport binary directly.
+
+## Success criteria
+Users are able to configure teleport to automatically create alerts in Opsgenie from access requests.
+Alerts were chosen over incidents in Opsgenie for high priority alerts that indicate a service interruption.
+Users are also able to configure auto approval flows to be met under certain conditions. E.g when a requester is on-call 
+
+## Configuration UX
+
+The plugin will be configured using a toml file containing the required information to interact with both Teleport access and the Opsgenie API.
```

Comment:
Let's try to avoid adding [two competing plugin configuration mechanisms](https://github.com/gravitational/teleport/pull/20990) in Teleport at the same time if possible.

Reply:
Agreed :+1:

Reply:
Hard agree. However, this needs some thought, as the scope of this configuration in hosted and standalone versions will be different:
* Hosted: we only configure what Cloud needs to authenticate with the provider (in OAuth case - client credentials; in OpsGenie case - might be nothing, since the user will need to supply a static API token). All the "settings knobs" are controled via:
  * `Plugin` resource which can be manipulated through UI (and UI-only for now at least),
  * Eventually, routing rules.
*  Self-hosted standalone: we need to configure Teleport identity & address, 3rd party provider auth, as well as "settings" (OpsGenie: priority, tags, etc.).